dts/arm/st/f1/stm32f105Xc.dtsi

```diff
@@ -18,5 +18,14 @@
 				reg = <0x08000000 DT_SIZE_K(256)>;
 			};
 		};
+
+		dac1: dac@40007400 {
+			compatible = "st,stm32-dac";
+			reg = <0x40007400 0x400>;
+			clocks = <&rcc STM32_CLOCK_BUS_APB1 0x20000000>;
+			status = "disabled";
+			label = "DAC_1";
+			#io-channel-cells = <1>;
+		};
```

Comment:
Move this to stm32f105.dtsi, this is still true (all stm32f105 SoCs have a DAC) and will also benefit to stm32f107.dtsi

Reply:
@erwango, isn't it the case that DAC is applicable only for

> This section applies to connectivity line, high-density and XL-density STM32F101xx and
STM32F103xx devices only.

and `stm32f100Xb` is considered to be "Medium-density device" (*"ranges between 64 and 128 Kbytes"*)?

Reply:
Added DAC to stm32f100Xb.dtsi and stm32f105.dtsi according to your suggestions